pkg/model/iam/iam_builder.go

```diff
@@ -773,6 +773,10 @@ func addECRPermissions(p *Policy) {
 		"ecr:DescribeRepositories",
 		"ecr:ListImages",
 		"ecr:BatchGetImage",
+		"ecr:ReplicateImage",
+		"ecr:BatchImportUpstreamImage",
+		"ecr:CreateRepository",
+		"ecr:TagResource",
```

Comment:
@rsafonseca Please move these to a separate `addECRPullThroughPermissions(p *Policy)` function and call it only when the flag is set.

Reply:
Kind ping on this @rsafonseca. Let me know if you have any feedback here.

Reply:
all done @hakman 👍